src/core/execute.c

```diff
@@ -215,6 +215,9 @@ static void exec_context_tty_reset(const ExecContext *context, const ExecParamet
 
         if (context->tty_vt_disallocate && path)
                 (void) vt_disallocate(path);
+
+        if (p && p->stdin_fd >= 0)
+                (void) terminal_set_size_fd(p->stdin_fd, context->tty_rows, context->tty_cols);
```

Comment:
we should probably do this before the vt_disallocate() above, since the TTY won't exist anymore if that worked.